src/game_battler.cpp

```diff
@@ -327,6 +327,40 @@ static bool non_permanent(int state_id) {
 	return Data::states[state_id - 1].type == RPG::State::Persistence_ends;
 }
 
+int Game_Battler::ApplyConditions()
+{
+	int damageTaken = 0;
+	std::vector<int16_t> inflictedStates = this->GetInflictedStates();
+	if (inflictedStates.size() != 0) {
+		for (int i = 0; i<inflictedStates.size(); ++i) {
+			RPG::State state = Data::states[inflictedStates[i]];
+			int hp = state.hp_change_val;
+			int sp = state.sp_change_val;
```

Comment:
This calculation ignores the "Percentage of max HP/SP". In other sections of code where this is required we do something like this:
```int hp = state.hp_change_val + (int)(std::ceil(GetMaxHp() * state.hp_change_max / 100.0));```